### PR TITLE
Fix Range error found with Issue 2485

### PR DESCRIPTION
**Pull request by hesingh**
See https://github.com/p4lang/p4c/issues/2485 and this error from it:

`Can not shift by a negative value.`

This issue is caused by using ffs() from gmputil.h which returns `-1` when v is 0.

### CODE REVIEW

File: lib/gmputil.h [hunk 1]
```diff
@@ -84,7 +84,7 @@ static inline unsigned bitcount(big_int v) {
 }
 
 static inline int ffs(big_int v) {
-    if (v <= 0) return -1;
+    if (v <= 0) return (v < 0) ? -1 : 0;
```
Comment (mbudiu-vmw) [inline lib/gmputil.h:87]: we should probably not change the definition of ffs; if all bits are 0 returning 1 seems incorrect. Instead probably you should move this check to the function that calls ffs.

**Comment by mbudiu-vmw** [comment]
Fixes #2485

File: testdata/p4_16_samples_outputs/issue2485-midend.p4 [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+parser p(packet_in packet, out Headers hdr) {
+    state start {
+        packet.extract<H>(hdr.h);
+        transition select(hdr.h.a) {
+            16w0x0 &&& 16w0xffff: parse;
+            16w0x1 &&& 16w0xffff: parse;
+            16w0x2 &&& 16w0xfffe: parse;
+            16w0x4 &&& 16w0xffff: parse;
```
Comment (jafingerhut) [inline testdata/p4_16_samples_outputs/issue2485-midend.p4:21]: This range can be represented using only 2 ternary value/masks, i.e. (16w0x0 &&& 16w0xfffc) and (16w0x4 &&& 16w0xffff).  It seems odd to represent it using more ternary value/masks than needed.

Comment (hesingh) [reply testdata/p4_16_samples_outputs/issue2485-midend.p4:21]: This is an issue when the start of Range is zero.  I will have to mull over this one.  Right now, the exhaustive test you wrote in range-to-tcam-entries.c is fine with four entries.

**Pull request merged**